src/modeler/AutoRest.Swagger/Model/ValidationUtilities.cs

```diff
@@ -170,5 +171,21 @@ public static bool IsReferenceOfType(string reference, Dictionary<string, Schema
 
             return false;
         }
+
+        /// <summary>
+        /// Returns array of resource providers
+        /// </summary>
+        /// <param name="paths">Dictionary of paths to look for</param>
+        /// <returns>Array of resource providers</returns>
+        public static string[] GetResourceProviders(Dictionary<string, Dictionary<string, Operation>> paths)
```

Comment:
nit: I prefer returning IEnumerables instead of arrays, allows more functionality